src/extras/simplePagination.ts

```diff
@@ -40,22 +40,36 @@ export const simplePagination = (params?: PaginationParams): Resolver => {
     const { parentKey: key, fieldName } = info;
     const connections = cache.resolveConnections(key, fieldName);
     const size = connections.length;
-    const result: string[] = [];
+    let result: NullArray<string> = [];
     const visited = new Set();
+    let prevOffset: number | null = null;
     if (size === 0) return undefined;
 
     for (let i = 0; i < size; i++) {
       const [args, linkKey] = connections[i];
       if (!compareArgs(fieldArgs, args)) continue;
       const links = cache.resolveValueOrLink(linkKey) as string[];
-      if (links.length === 0) return undefined;
-      result.push(
-        ...links.reduce<string[]>((acc, lk) => {
-          if (visited.has(lk)) return acc;
-          visited.add(lk);
-          return [...acc, lk];
-        }, [])
-      );
+      if (links === null || links.length === 0) continue;
+
+      if (!prevOffset || (args as any)[offsetArgument] > prevOffset) {
```

Comment:
I know this is more of a nit, but shall we assign `args[offsetArgument]` properly and add a check to the above `continue` branch that skips the connection if `typeof args[offsetArgument] !== 'number'`?